src/libraries/System.Runtime/tests/System/Reflection/ModuleTests.cs

```diff
@@ -41,6 +41,7 @@ namespace System.Reflection.Tests
 {
     public class ModuleTests
     {
+        public const string ConstField = "Value";
```

Comment:
This would cause `MissingFieldException` mentioned in the issue: https://github.com/dotnet/runtime/blob/1350579859435ee18bd530e18022a880ae418132/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeModule.cs#L222-L225 
And will be tested with existing test below https://github.com/dotnet/runtime/blob/de7d2f1dd2a4fd202608fefa7310e9d44e354dc1/src/libraries/System.Runtime/tests/System/Reflection/ModuleTests.cs#L303-L312

Reply:
Does it need to be a field on a Module, and not a Class?

Reply:
I guess `Module.ResolveField(fi.MetadataToken)` works for any field defined within a module, not only global fields. If I remove the `catch (MissingFieldException) ` catch block and add this method the test fails